pwnlib/heap/glmalloc/heap_explorer.py

```diff
@@ -0,0 +1,698 @@
+            =====... Fast Bins =====...
+            ...
+            ======...
+            >>> number_of_fastbins = len(fast_bins)
+            >>> fast_bins_counts = [len(fast_bin) for fast_bin in fast_bins]
+        """
+        malloc_state = self.malloc_state(arena_index)
+        return self._fast_bin_parser.parse_all_from_malloc_state(malloc_state)
+
+    def all_arenas_fast_bins(self):
+        """Returns the small bins of all arenas
+
+        Returns:
+            :obj:`list` of :class:`FastBins`
+
+        Example:
+            >>> p = process('sh')
+            >>> p.sendline('init bins')
+            >>> hp = p.heap_explorer()
+            >>> bb = hp.all_arenas_fast_bins()
+            >>> bb_str = "\\n".join([str(b) for b in bb])
+            >>> print(bb_str) # doctest: +ELLIPSIS
+            =====... Fast Bins =====...
+            ...
+        """
+        malloc_states = self.all_arenas_malloc_states()
+        all_fast_bins = []
+        for malloc_state in malloc_states:
+            all_fast_bins.append(
+                self._fast_bin_parser.parse_all_from_malloc_state(malloc_state)
+            )
+        return all_fast_bins
+
+    def tcaches(self, arena_index=0):
+        """Returns the tcaches of the arena
+
+        Args:
+            arena_index (int, optional): The index of the desired arena. If none
+                is specified, then the index of the main arena will be selected
+
+        Returns:
+            :class:`Tcaches`
+
+        Example:
+            >>> p = process('sh')
+            >>> p.sendline('init bins')
+            >>> hp = p.heap_explorer()
+            >>> try:
+            ...     tcaches_str = str(hp.tcaches())
+            ... except NoTcacheError:
+            ...     tcaches_str = ""
+            ...
+            >>> print(tcaches_str) # doctest: +SKIP
+            =================================== Tcaches ===================================
+            [23] Tcache 0x188 (1) => Chunk(0x56383e3c0250 0x190 PREV_IN_USE) => 0x0
+            [41] Tcache 0x2a8 (1) => Chunk(0x56383e3bffa0 0x2b0 PREV_IN_USE) => 0x0
+            ================================================================================
+
+        """
+        if not self.tcaches_enabled:
+            raise NoTcacheError()
+
+        malloc_state = self.malloc_state(arena_index)
+        return self._tcache_parser.parse_all_from_malloc_state(malloc_state)
+
+    def all_arenas_tcaches(self):
+        """Returns the tcaches of all arenas
+
+        Returns:
+            :obj:`list` of :class:`Tcaches`
+
+        Example:
+            >>> p = process('sh')
+            >>> p.sendline('init bins')
+            >>> hp = p.heap_explorer()
+            >>> try:
+            ...     ttcc = hp.all_arenas_tcaches()
+            ...     tcaches_str = "\\n".join([str(tc) for tc in ttcc])
+            ... except NoTcacheError:
+            ...     tcaches_str = ""
+            ...
+            >>> print(tcaches_str) # doctest: +SKIP
```

Comment:
Use a corefile so we don't have to skip these tests.  We want the tests to actually work.

Reply:
Done in 72d99e4